admin/class-boldgrid-backup-admin-archiver-cancel.php

```diff
@@ -0,0 +1,142 @@
+<?php
+/**
+ * File: class-boldgrid-backup-admin-archiver-cancel.php
+ *
+ * @link       https://www.boldgrid.com
+ * @since      SINCEVERSION
+ *
+ * @package    Boldgrid_Backup
+ * @copyright  BoldGrid
+ * @version    $Id$
+ * @author     BoldGrid <support@boldgrid.com>
+ */
+
+/**
+ * Class: Boldgrid_Backup_Admin_Archiver_Cancel
+ *
+ * @since SINCEVERSION
+ */
+class Boldgrid_Backup_Admin_Archiver_Cancel {
+	/**
+	 * Add cancelation data to the in progress data.
+	 *
+	 * @since SINCEVERSION
+	 */
+	public static function add_progress_data() {
+		Boldgrid_Backup_Admin_In_Progress_Data::set_arg( 'process_error', __( 'This backup has been successfully canceled by the user.', 'boldgrid-backup' ) );
+		Boldgrid_Backup_Admin_In_Progress_Data::set_arg( 'success', false );
+	}
+
+	/**
+	 * Whether or not we can cancel a backup.
+	 *
+	 * @since SINCEVERSION
+	 */
+	public static function can_cancel() {
+		$pid = Boldgrid_Backup_Admin_In_Progress_Data::get_arg( 'pid' );
+
+		if ( empty( $pid ) ) {
+			return false;
+		}
+
+		/*
+		 * Signal 0 does not actually get sent, but kill will check to see if it is possible to send
+		 * the signal.
+		 */
+		return posix_kill( $pid, 0 );
+	}
+
+	/**
+	 * Cancel a backup.
+	 *
+	 * @since SINCEVERSION
+	 *
+	 * @return bool
+	 */
+	public static function cancel() {
+		$pid = Boldgrid_Backup_Admin_In_Progress_Data::get_arg( 'pid' );
+		if ( empty( $pid ) ) {
+			return false;
+		}
+
+		if ( ! self::can_cancel() ) {
+			return false;
+		}
+
```

Comment:
If you add this, it should resolve the issue with canceling not working when pcntl_async_signals() is not defined.

```
if ( ! defined( 'SIGUSR1' ) ) {
	define( 'SIGUSR1', 10 );
}
```

Reply:
If you add this, it should resolve the issue with canceling not working when pcntl_async_signals() is not defined.